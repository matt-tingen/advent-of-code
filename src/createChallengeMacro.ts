import { getSolver } from '~/run';

export const createChallengeMacro = (year: number) =>
  createMacro(
    async (day: number, part: 'a' | 'b') => {
      const solver = await getSolver(year, day, part);
      const result = await solver();

      expect(result).toMatchSnapshot();
    },
    (provided, day, part) => provided || `Challenge ${year} ${day}${part}`,
  );
