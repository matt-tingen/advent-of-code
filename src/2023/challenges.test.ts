import { createChallengeMacro } from '~/createChallengeMacro';

const testChallenge = createChallengeMacro(2023);

run(testChallenge, 1, 'a');
run(testChallenge, 1, 'b');
run(testChallenge, 2, 'a');
run(testChallenge, 2, 'b');
run(testChallenge, 3, 'a');
run(testChallenge, 3, 'b');
run(testChallenge, 4, 'a');
run(testChallenge, 4, 'b');
run(testChallenge, 5, 'a');
run(testChallenge, 5, 'b');
run(testChallenge, 6, 'a');
run(testChallenge, 6, 'b');
run(testChallenge, 7, 'a');
run(testChallenge, 7, 'b');
run(testChallenge, 8, 'a');
run(testChallenge, 8, 'b');
run(testChallenge, 9, 'a');
run(testChallenge, 9, 'b');
