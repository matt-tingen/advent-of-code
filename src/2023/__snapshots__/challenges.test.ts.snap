// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Challenge 2023 1a 1`] = `53974`;

exports[`Challenge 2023 1b 1`] = `52840`;

exports[`Challenge 2023 2a 1`] = `2377`;

exports[`Challenge 2023 2b 1`] = `71220`;

exports[`Challenge 2023 3a 1`] = `525119`;

exports[`Challenge 2023 3b 1`] = `76504829`;

exports[`Challenge 2023 4a 1`] = `15205`;

exports[`Challenge 2023 4b 1`] = `6189740`;

exports[`Challenge 2023 5a 1`] = `309796150`;

exports[`Challenge 2023 5b 1`] = `50716416`;

exports[`Challenge 2023 6a 1`] = `633080`;

exports[`Challenge 2023 6b 1`] = `20048741`;

exports[`Challenge 2023 7a 1`] = `246409899`;

exports[`Challenge 2023 7b 1`] = `244848487`;

exports[`Challenge 2023 8a 1`] = `21409`;

exports[`Challenge 2023 8b 1`] = `21165830176709`;

exports[`Challenge 2023 9a 1`] = `1789635132`;

exports[`Challenge 2023 9b 1`] = `913`;

exports[`Challenge 2023 10a 1`] = `6725`;

exports[`Challenge 2023 10b 1`] = `383`;

exports[`Challenge 2023 11a 1`] = `10490062`;

exports[`Challenge 2023 11b 1`] = `382979724122`;

exports[`Challenge 2023 13a 1`] = `29130`;
